lib/compiler/passes/generate-bytecode.js

```diff
@@ -5,6 +5,56 @@ let js = require("../js");
 let op = require("../opcodes");
 let visitor = require("../visitor");
 
+// Determines if rule always used in disabled report failure context,
+// that means, that any failures, reported within it, are never will be
+// visible, so the no need to report it.
+function calcReportFailuresForRules(ast, allowedStartRules) {
```

Comment:
This pass is not extracted in the separate file because it strongly influences generation of a bytecode, and therefore shall be tested together with generation of a bytecode. Actually, it is just a detail of implementation of the bytecode generator